<launch>
    <arg name="namespace_prefix" default="kinova_realsense_calibrate" />
  
    <arg name="marker_size" default="0.10" doc="Size of the ArUco marker used, in meters" />
    <arg name="marker_id"   default="10" doc="The ID of the ArUco marker used" />

    <arg name="camera_"      value="camera_on_hand" />
    <arg name="serial_no"    value=""/>

    <arg name="robot_type" value="m1n6s200"/>

    <arg name="eye_on_hand" value="true"/>

    <!-- start the Kinect -->



    <include file="$(find realsense2_camera)/launch/rs_camera.launch" >
       
        <arg name="camera"        value="$(arg camera_)"/>
        <arg name="serial_no"          value="$(arg serial_no)"/>
    </include>




    <include file="$(find kinova_bringup)/launch/kinova_robot.launch">
        <param name="kinova_robotType" value="$(arg robot_type)"/>
    </include>


     <include file="$(find m1n6s200_moveit_config)/launch/$(arg robot_type)_demo.launch" />

    

    <node name="aruco_tracker" pkg="aruco_ros" type="single">
        <remap from="/camera_info" to="$(arg camera_)/color/camera_info" />
        <remap from="/image" to="$(arg camera_)/color/image_raw" />
        <param name="image_is_rectified" value="true"/>
        <param name="marker_size"        value="$(arg marker_size)"/>
        <param name="marker_id"          value="$(arg marker_id)"/>
        <param name="reference_frame"    value="$(arg camera_)_link"/>
        <param name="camera_frame"       value="$(arg camera_)_color_optical_frame"/>
        <param name="marker_frame"       value="camera_marker" />
    </node>
 

  
    <include file="$(find easy_handeye)/launch/calibrate.launch" >
        <arg name="namespace_prefix" value="$(arg namespace_prefix)" />
        <arg name="eye_on_hand" value="$(arg eye_on_hand)" />

        <arg name="tracking_base_frame" value="$(arg camera_)_link" />
        <arg name="tracking_marker_frame" value="camera_marker" />
        <arg name="robot_base_frame" value="$(arg robot_type)_link_base" />

        <arg name="robot_effector_frame" value="$(arg robot_type)_end_effector" />


        <arg name="freehand_robot_movement" value="false" />
        <arg name="robot_velocity_scaling" value="0.15" />
        <arg name="robot_acceleration_scaling" value="0.1" />
    </include>



</launch>
